### ISSUE TITLE
Liquidsoap escapes apostrophes in track titles in metadata from the telnet connection.

### ISSUE THREAD

**Issue by oranjebloom** [savonet/liquidsoap#2732]
I am using version 2.1.2 on Debian 10 installed via opam. 

I have noticed that Liquidsoap escapes apostrophes in file names and track titles in metadata from the telnet connection.

For example: 
```
filename="/ssr/ssr1/pax/general/Throwing Snow - Melum (Heathered Pearls\' dormancy remix).mp3"

title="Melum (Heathered Pearls\' dormancy remix)"
```
The ID3 tags for these tracks do not contain a backslash before the apostrophe. I’d like to know why Liquidsoap escapes the apostrophes? Also, which other characters (if any) are escaped?

**Comment by toots** [comment]
Hi,

Can you describe more in details in what circumstances you see these apostrophes being escaped?

String representation of sequence of characters require escaping the quotation marker. We support two types of quotation markers, single quote `'` and double quote `"`.

When reading strings, the convention is to escape these characters (and other, see: https://www.liquidsoap.info/doc-2.1.2/language.html)

So, the string `bla'foo` can be written: `'bla\'foo'`.

By convention, we also accept escaping single quotes inside a double-quoted string, so this is also valid: `"bla\'foo"`.

When printing script values on logs and other places, we apply escaping to reproduce how a string would have to be entered to be correct. It's possible that we do indeed escape single quotes even though the string is double-quoted.

Is that such a situation? In this case, while perhaps not elegant, this is still a valid representation and should not impact the internal tags/filenames etc.

**Comment by oranjebloom** [comment]
@toots I am using the metadata from the telnet connection to create a list of 10 most recently played tracks on the radio webpage at [systrum.net](https://systrum.net ). If the artist or track title has an apostrophe in the id3 tag for that file , the metadata is displayed with a slash before the apostrophe in Liquidsoap (and consequently on my webpage also) as shown in the attached screen shots. 

As a work-around, in the website's code, I have to replace \' with ' and \" with " 
So it's like this in the code:

replace "\'" with "'"
replace '\"' with '"' 

This solution works, and as long as only these characters are being escaped by Liquidsoap, it's fine.

![telnet metadata](https://user-images.githubusercontent.com/46059094/201516621-9f44b695-99ab-4223-94a8-02712c9fddb0.jpg)

![recently played list](https://user-images.githubusercontent.com/46059094/201516630-8d2d1db0-6630-419b-9618-a20aa496e015.jpg)

**Comment by toots** [comment]
Thanks for the details. This should be fixed in `rolling-release-v2.1.x` and `main` now.
